src/main/java/org/jabref/gui/fieldeditors/IdentifierEditorViewModel.java

```diff
@@ -53,21 +56,36 @@ public IdentifierEditorViewModel(Field field, SuggestionProvider<?> suggestionPr
         this.preferences = preferences;
         this.field = field;
 
-        identifier.bind(
-                EasyBind.map(text, input -> IdentifierParser.parse(field, input))
-        );
-
-        validIdentifierIsNotPresent.bind(
-                EasyBind.map(identifier, parsedIdentifier -> parsedIdentifier.isEmpty())
-        );
-
         idFetcherAvailable.setValue(WebFetchers.getIdFetcherForField(field).isPresent());
     }
 
     public boolean isIdFetcherAvailable() {
         return idFetcherAvailable.get();
     }
 
+    @Override
+    public void bindToEntry(BibEntry entry) {
+        super.bindToEntry(entry);
+        // Unlike other identifiers (they only depend on their own field value), eprint  depends on eprinttype thus it
+        // needs to be reconstructed everytime the eprinttype field changes.
+        if (field == StandardField.EPRINT) {
+            entry.getFieldsObservable().addListener((MapChangeListener<Field, String>) change -> {
+                Field changedField = change.getKey();
+                if (StandardField.EPRINTTYPE.equals(changedField)) {
+                    this.identifierParser = new IdentifierParser(entry);
+                    identifier.bind(EasyBind.map(text, input -> identifierParser.parse(field, input)));
+                    validIdentifierIsNotPresent.bind(EasyBind.map(identifier, Optional::isEmpty));
+                }
+            });
+        }
+
+        this.identifierParser = new IdentifierParser(entry);
+
+        identifier.bind(EasyBind.map(text, input -> identifierParser.parse(field, input)));
+
+        validIdentifierIsNotPresent.bind(EasyBind.map(identifier, Optional::isEmpty));
+    }
```

Comment:
I think this method will be called every time the editor is created, and so is the listener, and I think that this will probably lead to unexepected behaviour. Im not sure if this can be put somewhere else or the listener be removed before closing...

Reply:
FYI: this class will be removed once I refactor the tests for the identifier view model. But the listener can still leak in EprintIdentifierEditorViewModel#bindToEntry. Don't have any idea how to prevent it though... Adding 'unbindFromEntry' could one part of the solution.

Reply:
I found a bigger leak in the entry editor, the number of invalidation listeners for an entry increases by 47 every time that entry is selected *(For some reason it stopped growing at 709)*. My leak only grows by 1... Doesn't sound so horrible now does it :^)

![image](https://user-images.githubusercontent.com/21198231/216847028-82e095ea-9326-4b82-b88d-d7c474a8bc23.png)

Reply:
🤣 🤣 🤣 ... ... 😭

Reply:
Was laughing so hard upon reading this 🤣